File: datasets/C++/Gemini/DataStructures/medium1.cpp
#include <iostream> // Required for input/output operations (cin, cout)
#include <vector>   // Required for using std::vector
#include <string>   // Required for using std::string

// Function to count occurrences of query strings
std::vector<int> matchingStrings(const std::vector<std::string>& stringList, const std::vector<std::string>& queries) {
    std::vector<int> results;
    for (const std::string& query : queries) {
        int count = 0;
        for (const std::string& s : stringList) {
            if (s == query) {
                count++;
            }
        }
        results.push_back(count);
    }
    return results;
}

int main() {
    // --- Input section ---
    int n;
    std::cout << "Enter the number of strings in stringList (n):" << std::endl;
    std::cin >> n;

    std::cout << "Enter " << n << " strings for stringList, one per line:" << std::endl;
    std::vector<std::string> stringList(n);
    for (int i = 0; i < n; ++i) {
        std::cin >> stringList[i];
    }

    int q;
    std::cout << "\nEnter the number of query strings (q):" << std::endl;
    std::cin >> q;

    std::cout << "Enter " << q << " query strings, one per line:" << std::endl;
    std::vector<std::string> queries(q);
    for (int i = 0; i < q; ++i) {
        std::cin >> queries[i];
    }

    // --- Processing and Output section ---
    std::vector<int> results = matchingStrings(stringList, queries);

    std::cout << "\n--- Results ---" << std::endl;
    for (int res : results) {
        std::cout << res << std::endl;
    }

    return 0;
}

File: datasets/Python/Gemini/DataStructures/medium1.py
def matchingStrings(stringList, queries):
    """
    Counts occurrences of each query string in a list of input strings.

    Args:
        stringList: A list of strings to search within.
        queries: A list of query strings to count.

    Returns:
        A list of integers, where each element is the count of the
        corresponding query string in stringList.
    """
    results = []
    for query in queries:
        count = 0
        for s in stringList:
            if s == query:
                count += 1
        results.append(count)
    return results

if __name__ == "__main__":
    # --- Input section ---
    print("Enter the number of strings in stringList (n):")
    n = int(input())

    print(f"Enter {n} strings for stringList, one per line:")
    stringList = []
    for _ in range(n):
        stringList.append(input())

    print("\nEnter the number of query strings (q):")
    q = int(input())

    print(f"Enter {q} query strings, one per line:")
    queries = []
    for _ in range(q):
        queries.append(input())

    # --- Processing and Output section ---
    print("\n--- Results ---")
    results = matchingStrings(stringList, queries)
    for res in results:
        print(res)
